process/transactionProcessor.go

```diff
@@ -455,19 +469,46 @@ func (tp *TransactionProcessor) getTxFromObservers(txHash string, reqType reques
 	return nil, errors.ErrTransactionNotFound
 }
 
-func (tp *TransactionProcessor) getTxWithSenderAddr(txHash, sender string) (*data.FullTransaction, error) {
+func (tp *TransactionProcessor) alterTxWithScResultsFromSourceIfNeeded(txHash string, tx *data.FullTransaction, withEvents bool) *data.FullTransaction {
+	if !withEvents && len(tx.ScResults) == 0 {
+		return tx
+	}
+
+	observers, _, err := tp.getNodesInShard(tx.SourceShard, requestTypeFullHistoryNodes)
+	if err != nil {
+		return tx
+	}
+
+	for _, observer := range observers {
+		getTxResponse, ok, _ := tp.getTxFromObserver(observer, txHash, withEvents)
+		if !ok {
+			continue
+		}
+
+		alteredTxFromDest := mergeScResultsFromSourceAndDestIfNeeded(&getTxResponse.Data.Transaction, tx, withEvents)
+		return alteredTxFromDest
+	}
+
+	return tx
+}
+
+func (tp *TransactionProcessor) getTxWithSenderAddr(txHash, sender string, withEvents bool) (*data.FullTransaction, error) {
 	sndShardID, err := tp.getShardByAddress(sender)
 	if err != nil {
 		return nil, errors.ErrInvalidSenderAddress
 	}
 
-	observers, err := tp.proc.GetObservers(sndShardID)
+	observers, haveFullHistoryObservers, err := tp.getNodesInShard(sndShardID, requestTypeFullHistoryNodes)
 	if err != nil {
 		return nil, err
 	}
 
+	// if we dont have full history observers withEvents needs to be false because only full history observers can
```

Comment:
As above, do you think you can remove this arbitrary constraint, for the moment (so that we don't break the Squad installers and so on)?

Reply:
removed.